contrib/actions/update-modules.sh

```diff
@@ -1,39 +1,68 @@
-#!/bin/bash
-set -eEu
-set -o pipefail
+#!/usr/bin/env bash
 
-# file that contains the module information
-MODULES_FILE="modules"
+set -eo pipefail
 
-if ((BASH_VERSINFO[0] < 4)); then
-    echo "This script requires Bash 4.0 or above."
-    exit 1
-fi
+# move to basedir, in case the script is not executed via `make update-modules`
+cd "../../" || exit 1
 
-function get_value_from_line() {
-    cat | cut -d '=' -f 2
-}
+# shellcheck source=./modules
+source ./modules
 
-function get_all_repo_names() {
-    local file=$1
-    grep "^GLUON_SITE_FEEDS" "${file}" | get_value_from_line | tr -d "'"
+git diff --quiet ./modules || {
+	1>&2 echo "Your modules file is dirty, aborting."
+	exit 1
 }
 
-for repo in $(get_all_repo_names "${MODULES_FILE}"); do
-    REPO_URL=$(grep "^PACKAGES_${repo^^}_REPO" "${MODULES_FILE}" | get_value_from_line)
-    REPO_COMMIT=$(grep "^PACKAGES_${repo^^}_COMMIT" "${MODULES_FILE}" | get_value_from_line)
-    REPO_BRANCH=$(grep "^PACKAGES_${repo^^}_BRANCH" "${MODULES_FILE}" | get_value_from_line)
+LOCAL_BRANCH=$(git branch --show-current)
+[[ $LOCAL_BRANCH != update-modules-* ]] && LOCAL_BRANCH="update-modules-$LOCAL_BRANCH-$(date +%s)"
+
+for MODULE in "PACKAGES_COMMUNITY"; do
```

Comment:
atm only on legacy we have more packages Repo than community

Reply:
Can't you just use the beautiful loop we had before?
```sh
...
for repo in $(get_all_repo_names "${MODULES_FILE}"); do
...
```

Reply:
Found a much better solution 🙂